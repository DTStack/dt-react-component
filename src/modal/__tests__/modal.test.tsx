import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { alert, modal } from 'ant-design-testing';
import '@testing-library/jest-dom';

import Modal from '../';

describe('Test Modal Component', () => {
    beforeEach(() => {
        cleanup();
    });

    it('Should Match snapshot', () => {
        const { asFragment } = render(
            <Modal visible title="title" getContainer={false}>
                <div>test</div>
            </Modal>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('Should get different size', () => {
        // default size
        const { container, rerender } = render(
            <Modal visible title="title" getContainer={false}>
                test
            </Modal>
        );
        expect(modal.query(container)?.style.width).toBe('640px');

        // small size
        rerender(
            <Modal visible title="title" getContainer={false} size="small">
                test
            </Modal>
        );
        expect(modal.query(container)?.style.width).toBe('400px');

        // large size
        rerender(
            <Modal visible title="title" getContainer={false} size="large">
                test
            </Modal>
        );
        expect(modal.query(container)?.style.width).toBe('900px');
    });

    describe('Should support banner', () => {
        it('Should support string banner', async () => {
            const { getByText } = render(
                <Modal visible title="title" getContainer={false} banner="banner">
                    test
                </Modal>
            );

            expect(getByText('banner')).toBeInTheDocument();
        });

        it('Should support ReactNode', async () => {
            const { getByTestId } = render(
                <Modal
                    visible
                    title="title"
                    getContainer={false}
                    banner={<div data-testid="banner">xxxx</div>}
                >
                    test
                </Modal>
            );

            expect(getByTestId('banner')).toBeInTheDocument();
        });

        it('Should support AlertProps', async () => {
            const { container, getByText } = render(
                <Modal
                    visible
                    title="title"
                    getContainer={false}
                    banner={{ message: 'banner', type: 'error' }}
                >
                    test
                </Modal>
            );

            expect(getByText('banner')).toBeInTheDocument();
            expect(alert.query(container)?.classList.contains('ant-alert-error')).toBeTruthy();
        });
    });
});
