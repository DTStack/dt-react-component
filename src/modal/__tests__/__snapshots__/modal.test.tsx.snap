// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Modal Component Should Match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-modal-root"
  >
    <div
      class="ant-modal-mask"
    />
    <div
      class="ant-modal-wrap"
      tabindex="-1"
    >
      <div
        aria-labelledby="test-id"
        aria-modal="true"
        class="ant-modal dtc-modal"
        role="dialog"
        style="height: auto; width: 520px;"
      >
        <div
          aria-hidden="true"
          style="width: 0px; height: 0px; overflow: hidden; outline: none;"
          tabindex="0"
        />
        <div
          class="ant-modal-content"
        >
          <button
            aria-label="Close"
            class="ant-modal-close"
            type="button"
          >
            <span
              class="ant-modal-close-x"
            >
              <span
                aria-label="close"
                class="anticon anticon-close ant-modal-close-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close"
                  fill="currentColor"
                  fill-rule="evenodd"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                  />
                </svg>
              </span>
            </span>
          </button>
          <div
            class="ant-modal-header"
          >
            <div
              class="ant-modal-title"
              id="test-id"
            >
              title
            </div>
          </div>
          <div
            class="ant-modal-body"
            style="padding: 0px;"
          >
            <section
              class="dtc-modal-body"
            >
              <div>
                test
              </div>
            </section>
          </div>
          <div
            class="ant-modal-footer"
          >
            <button
              class="ant-btn ant-btn-default"
              type="button"
            >
              <span>
                Cancel
              </span>
            </button>
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                OK
              </span>
            </button>
          </div>
        </div>
        <div
          aria-hidden="true"
          style="width: 0px; height: 0px; overflow: hidden; outline: none;"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test Modal Component Should support banner Should match snapshot for draggable modal 1`] = `
<DocumentFragment>
  <div
    class="ant-modal-root"
  >
    <div
      class="ant-modal-mask"
    />
    <div
      class="ant-modal-wrap"
      tabindex="-1"
    >
      <div
        aria-labelledby="test-id"
        aria-modal="true"
        class="ant-modal dtc-modal dtc-modal__draggable"
        role="dialog"
        style="height: auto; width: 520px;"
      >
        <div
          aria-hidden="true"
          style="width: 0px; height: 0px; overflow: hidden; outline: none;"
          tabindex="0"
        />
        <div
          class="dtc-float-container react-draggable"
          style="width: 520px; height: auto; transform: translate(0px,0px);"
        >
          <div
            class="ant-modal-content"
          >
            <button
              aria-label="Close"
              class="ant-modal-close"
              type="button"
            >
              <span
                class="ant-modal-close-x"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close ant-modal-close-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    fill-rule="evenodd"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                title
              </div>
            </div>
            <div
              class="ant-modal-body"
              style="padding: 0px;"
            >
              <section
                class="dtc-modal-body"
              >
                test
              </section>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn ant-btn-default"
                type="button"
              >
                <span>
                  Cancel
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  OK
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          style="width: 0px; height: 0px; overflow: hidden; outline: none;"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test Modal Component Should support banner Should match snapshot for resizable modal 1`] = `
<DocumentFragment>
  <div
    class="ant-modal-root"
  >
    <div
      class="ant-modal-mask"
    />
    <div
      class="ant-modal-wrap"
      tabindex="-1"
    >
      <div
        aria-labelledby="test-id"
        aria-modal="true"
        class="ant-modal dtc-modal dtc-modal__resizable"
        role="dialog"
        style="height: 600px; width: 620px;"
      >
        <div
          aria-hidden="true"
          style="width: 0px; height: 0px; overflow: hidden; outline: none;"
          tabindex="0"
        />
        <div
          class="ant-modal-content"
        >
          <button
            aria-label="Close"
            class="ant-modal-close"
            type="button"
          >
            <span
              class="ant-modal-close-x"
            >
              <span
                aria-label="close"
                class="anticon anticon-close ant-modal-close-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close"
                  fill="currentColor"
                  fill-rule="evenodd"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                  />
                </svg>
              </span>
            </span>
          </button>
          <div
            class="ant-modal-header"
          >
            <div
              class="ant-modal-title"
              id="test-id"
            >
              title
            </div>
          </div>
          <div
            class="ant-modal-body"
            style="padding: 0px;"
          >
            <section
              class="dtc-modal-body"
            >
              test
            </section>
          </div>
          <div
            class="ant-modal-footer"
          >
            <button
              class="ant-btn ant-btn-default"
              type="button"
            >
              <span>
                Cancel
              </span>
            </button>
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                OK
              </span>
            </button>
          </div>
        </div>
        <div
          class="dt-modal-resize-handle handle-s"
        />
        <div
          class="dt-modal-resize-handle handle-w"
        />
        <div
          class="dt-modal-resize-handle handle-e"
        />
        <div
          class="dt-modal-resize-handle handle-n"
        />
        <div
          class="dt-modal-resize-handle handle-ne"
        />
        <div
          class="dt-modal-resize-handle handle-nw"
        />
        <div
          class="dt-modal-resize-handle handle-sw"
        />
        <div
          class="dt-modal-resize-handle handle-se"
        />
        <div
          aria-hidden="true"
          style="width: 0px; height: 0px; overflow: hidden; outline: none;"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
